### DISCUSSION TITLE
Router address

### DISCUSSION THREAD

**Discussion by Antro24** [rejetto/hfs#685] [Q&A] [answered]
My router works at the address 10.0.0.1 and the additional one at the address 10.0.0.3
HFS detects my additional router at the address 10.0.0.3, although there is no cable connection with it, it works as an additional device without its own settings.

**Comment by rejetto** [comment]
i didn't understand the topology of your network.
what is your hfs-computer connected to?

**Reply by Antro24** [reply]
My HFS computer is connected to 10.0.0.1
But for some reason HFS climbs to 10.0.0.3
I disabled the additional router and realized that UpNp was not enabled in the main Mikrotik 10.0.0.1 router, I enabled this function and HFS was able to open the internal port successfully. Everything would be fine, but... I connected the 10.0.0.3 router back and restarted HFS, the program determined the router address back as 10.0.0.3. Conclusion: it is necessary to manually specify the router address, but such a setting is missing.


(All text has been translated via Google Translate.)

**Comment by rejetto** [comment]
hfs gives priority to what upnp says about who's the internet-gateway.
only when upnp doesn't respond, then it asks the system.
i guess 10.3 is presenting itself as such.
is hfs on windows, linux or mac?

**Reply by Antro24** [reply]
HFS runs on Windows

My additional router 10.0.0.3 is turned on as an access point, that is, it doesn’t even have a DHCP server, NAT, UPNP, nothing at all, it just distributes wifi.
And defining it as a gateway is wrong.

**Comment by rejetto** [comment]
can you copy here the content of the command "route print", executed in cmd?

**Comment by rejetto** [comment]
no, i meant you open cmd, like this
<img width="216" alt="image" src="https://github.com/user-attachments/assets/3eb52d30-7d96-4bff-a674-d3e14a522490">

then type "route print", hit ENTER, and copy the output

**Reply by Antro24** [reply]
Translate into your language.
```
===========================================================================
Список интерфейсов
 12...08 3a 88 6e ea df ......Realtek PCIe GbE Family Controller
  1...........................Software Loopback Interface 1
===========================================================================

IPv4 таблица маршрута
===========================================================================
Активные маршруты:
Сетевой адрес           Маска сети      Адрес шлюза       Интерфейс  Метрика
          0.0.0.0          0.0.0.0         10.0.0.1         10.0.0.5     25
         10.0.0.0    255.255.255.0         On-link          10.0.0.5    281
         10.0.0.5  255.255.255.255         On-link          10.0.0.5    281
       10.0.0.255  255.255.255.255         On-link          10.0.0.5    281
        127.0.0.0        255.0.0.0         On-link         127.0.0.1    331
        127.0.0.1  255.255.255.255         On-link         127.0.0.1    331
  127.255.255.255  255.255.255.255         On-link         127.0.0.1    331
        224.0.0.0        240.0.0.0         On-link         127.0.0.1    331
        224.0.0.0        240.0.0.0         On-link          10.0.0.5    281
  255.255.255.255  255.255.255.255         On-link         127.0.0.1    331
  255.255.255.255  255.255.255.255         On-link          10.0.0.5    281
===========================================================================
Постоянные маршруты:
  Отсутствует

IPv6 таблица маршрута
===========================================================================
Активные маршруты:
 Метрика   Сетевой адрес            Шлюз
  1    331 ::1/128                  On-link
  1    331 ff00::/8                 On-link
===========================================================================
Постоянные маршруты:
  Отсутствует

**Comment by rejetto** [comment, marked as answer]
ok, thanks.
this line reports the correct gateway
<img width="365" alt="image" src="https://github.com/user-attachments/assets/2ef3eb6e-6efe-4429-9d10-ebc8aee0cbe2">

So i guess the solution here is for HFS to not trust upnp about that, but alway ask Windows.

**Comment by Antro24** [comment]
I also haven’t found a built-in autorun function when the system starts.

**Comment by rejetto** [comment]
hfs is not providing this as a feature, at the moment

**Comment by rejetto** [comment]
version 0.54.0-alpha1 is including this fix
